tls/s2n_kem.c

```diff
@@ -217,15 +204,15 @@ const struct s2n_kem_group s2n_secp256r1_kyber_512_r2 = {
 };
 
 #if EVP_APIS_SUPPORTED
-const struct s2n_kem_group s2n_x25519_sike_p434_r2 = {
+const struct s2n_kem_group s2n_x25519_sike_p434_r3 = {
         .name = "x25519_sike-p434-r2",
```

Comment:
I'm debating whether we should change the name or not. I think `x25519_sike-p434-r2r3` might be clearer that the KEM Group applies to both versions. Same goes for the variable name.